internal/extproc/server.go

```diff
@@ -251,3 +298,17 @@ func filterSensitiveBody(resp *extprocv3.ProcessingResponse, logger *slog.Logger
 	}
 	return filteredResp
 }
+
+func getHeader(headers *corev3.HeaderMap, name string) string {
```

Comment:
nit: so basically i see we (unnecessarily) iterate through all headers here and headersToMap. would it be possible to use headersToMap before the processor selection and pass the constructed map to `processorForPath ` and eventually to `newProcessor`?

Reply:
I thought about this, but was unsure about modifying the processor factory methods to accept a request headers map. I can change that if you feel it is OK